### PR TITLE
Chat Info Refactor

### PR DESCRIPTION
**Pull request by MatkoTom**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant
motivation and context. List any dependencies that are required for this change.

Removed some unneeded observers and merged them all into one live data call.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration. All testing was done
manually by following, plainly, functionality requirements. If there are some special edge cases or
special ways that things need to be tested they will be mentioned below.

Dev testing

**Test Configuration**:

* Firmware version:
* Hardware: SAMSUNG Galaxy S22
* SDK: Android 14

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat/ChatMessagesFragment.kt [hunk 1]
```diff
@@ -494,6 +494,37 @@ class ChatMessagesFragment : BaseFragment(), ServiceConnection {
...
+                        }
+
+                        if (roomWithUsers?.room?.avatarFileId != room.responseData?.room?.avatarFileId) {
+                            room.responseData?.room?.avatarFileId?.let { avatarId ->
+                                setUserAvatar(avatarFileId = avatarId)
+                                roomWithUsers?.room?.avatarFileId = avatarId
+                            }
+                        }
+
+                        bindingSetup.chatHeader.tvTitle.text =
```
Comment (ivolarevic) [inline app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat/ChatMessagesFragment.kt:515]: Maybe you can also add if statement here to check if group number is different from previous?

Comment (MatkoTom) [reply app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat/ChatMessagesFragment.kt:515]: Users in RoomWithUsers is a val and cannot be reassigned currently. I can change it to var if you deem that to be alright?

Comment (ivolarevic) [reply app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat/ChatMessagesFragment.kt:515]: Maybe to check something like this: room.responseData?.users?.size != roomWithUsers.users.size?

Comment (MatkoTom) [reply app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat/ChatMessagesFragment.kt:515]: Yes, that is the idea, but if you don't change roomWithUsers.users yourself (similar to what we do with the avatar_file_id and group name), it will always hold it's old value. 

i.e. I have two users and add one more. The check would pass and the group would update correctly, but roomWithUsers.users.size would still remain at value 2. Hence, every room update would pass this test, even though no room users were added/removed and it could bug out if I return to user count 2 by removing someone.

**Review by ivolarevic — APPROVED** [review]
LGTM
